-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-47683/44228

ERROR: Build errors found during clang-tidy run.
```
--
src/L1Trigger/HardwareValidation/src/L1Comparator.cc:514:18: error: incompatible pointer types assigning to 'const L1MuRegionalCandCollection *' (aka 'const vector<L1MuRegionalCand> *') from 'L1MuGMTCandCollection *' (aka 'vector<L1MuGMTCand> *') [clang-diagnostic-error]
  514 |   gmt_can_data = &gmt_can_data_vec;
      |                  ^~~~~~~~~~~~~~~~~
src/L1Trigger/HardwareValidation/src/L1Comparator.cc:515:3: error: use of undeclared identifier 'gmt_can_emul'; did you mean 'gmt_rdt_emul'? [clang-diagnostic-error]
  515 |   gmt_can_emul = &gmt_can_emul_vec;
      |   ^~~~~~~~~~~~
      |   gmt_rdt_emul
--
src/L1Trigger/HardwareValidation/src/L1Comparator.cc:515:18: error: incompatible pointer types assigning to 'const L1MuRegionalCandCollection *' (aka 'const vector<L1MuRegionalCand> *') from 'L1MuGMTCandCollection *' (aka 'vector<L1MuGMTCand> *') [clang-diagnostic-error]
  515 |   gmt_can_emul = &gmt_can_emul_vec;
      |                  ^~~~~~~~~~~~~~~~~
src/L1Trigger/HardwareValidation/src/L1Comparator.cc:702:36: error: use of undeclared identifier 'gmt_can_data'; did you mean 'gmt_data'? [clang-diagnostic-error]
  702 |     process<L1MuGMTCandCollection>(gmt_can_data, gmt_can_emul, GMT, GMTcnd, eventInfo);
      |                                    ^~~~~~~~~~~~
      |                                    gmt_data
--
src/L1Trigger/HardwareValidation/src/L1Comparator.cc:702:50: error: use of undeclared identifier 'gmt_can_emul'; did you mean 'gmt_emul'? [clang-diagnostic-error]
  702 |     process<L1MuGMTCandCollection>(gmt_can_data, gmt_can_emul, GMT, GMTcnd, eventInfo);
      |                                                  ^~~~~~~~~~~~
      |                                                  gmt_emul
--
gmake: *** [config/SCRAM/GMake/Makefile.coderules:129: code-checks] Error 2
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

--- a/L1Trigger/DTBti/src/DTBtiChipEquations.cc
+++ b/L1Trigger/DTBti/src/DTBtiChipEquations.cc
@@ -160,8 +160,6 @@ void DTBtiChip::computeEqs() {
     //patt 0 always uses Xbc, because Xad could be negative...
     //when wire B is missing,  TshiftB=0
     //when hit B is gone out of shift register, Tshift=K0+1
-    float _difs_p0 = _difs[5];
-
     float TshiftB = 0;
     float TshiftC = 0;
     if (_thisStepUsedHit[3 - 1] == nullptr) {
@@ -180,7 +178,7 @@ void DTBtiChip::computeEqs() {
     } else
       TshiftC = _thisStepUsedHit[2 - 1]->jtrig();
 
-    _difs_p0 = (float)(TshiftB - TshiftC);
+    float _difs_p0 = (float)(TshiftB - TshiftC);
 
     // Zotto's
     _XeqAB_patt0 = (_sums[1] - K0) / 4.;  //eq. AB

--- a/L1Trigger/HardwareValidation/src/L1Comparator.cc
+++ b/L1Trigger/HardwareValidation/src/L1Comparator.cc
@@ -436,8 +436,6 @@ void L1Comparator::produce(edm::StreamID, edm::Event& iEvent, const edm::EventSe
   edm::Handle<L1MuGMTCandCollection> gmt_emul;
   edm::Handle<L1MuGMTReadoutCollection> gmt_rdt_data_;
   edm::Handle<L1MuGMTReadoutCollection> gmt_rdt_emul_;
-  L1MuRegionalCandCollection const* gmt_rdt_data = nullptr;
-  L1MuRegionalCandCollection const* gmt_rdt_emul = nullptr;
   if (runDoSys[GMT]) {
     iEvent.getByToken(tokenMuGMTCand_[0], gmt_data);
     iEvent.getByToken(tokenMuGMTCand_[1], gmt_emul);
@@ -509,10 +507,10 @@ void L1Comparator::produce(edm::StreamID, edm::Event& iEvent, const edm::EventSe
       gmt_rdt_emul_vec.insert(gmt_rdt_emul_vec.end(), rmc.begin(), rmc.end());
     }
   }
-  gmt_rdt_data = &gmt_rdt_data_vec;
-  gmt_rdt_emul = &gmt_rdt_emul_vec;
-  gmt_can_data = &gmt_can_data_vec;
-  gmt_can_emul = &gmt_can_emul_vec;
+  L1MuRegionalCandCollection const* gmt_rdt_data = &gmt_rdt_data_vec;
+  L1MuRegionalCandCollection const* gmt_rdt_emul = &gmt_rdt_emul_vec;
+  L1MuGMTCandCollection const* gmt_can_data = &gmt_can_data_vec;
+  L1MuGMTCandCollection const* gmt_can_emul = &gmt_can_emul_vec;
 
   ///--- done getting collections. ---
 

--- a/L1Trigger/L1CaloTrigger/plugins/L1NNCaloTauEmulator.cc
+++ b/L1Trigger/L1CaloTrigger/plugins/L1NNCaloTauEmulator.cc
@@ -531,6 +531,7 @@ void L1NNCaloTauEmulator::produce(edm::Event& iEvent, const edm::EventSetup& eSe
 
     float idScore = -1.;
     if (!VsPuId.method().empty()) {
+      VsPuId.passID(*cl3dIt, cluster);
       idScore = cluster.egVsPUMVAOut();
     }
 

